src/coreclr/jit/importer.cpp

```diff
@@ -4007,6 +4013,29 @@ GenTree* Compiler::impIntrinsic(GenTree*                newobjThis,
                 break;
             }
 
+            case NI_System_Runtime_CompilerServices_RuntimeHelpers_IsKnownConstant:
+            {
+                if (opts.OptimizationEnabled())
```

Comment:
Why do we need to check `OptimizationEnabled` here? We have have early out for disabled optimizations before this switch.

Reply:
The unnecessary `OptimizationEnabled` check is in a few other intrinsics as well. You can delete it there too while you are on it.